code/go/0chain.net/smartcontract/dbs/event/sharder.go

```diff
@@ -89,8 +85,37 @@ func (edb *EventDb) GetSharder(id string) (Sharder, error) {
 		First(&sharder).Error
 }
 
-func (edb *EventDb) GetShardersFromQuery(query *Sharder) ([]Sharder, error) {
+func (edb *EventDb) GetSharderWithDelegatePools(id string) (Sharder, []DelegatePool, error) {
+	var sharderDps []struct {
+		Sharder
+		DelegatePool
+		ProviderRewards //nolint
+	}
+	var s Sharder
+	var dps []DelegatePool
 
+	result := edb.Get().
+		Table("sharders").
+		Joins("left join provider_rewards on sharders.id = provider_rewards.provider_id").
+		Joins("left join delegate_pools on sharders.id = delegate_pools.provider_id").
```

Comment:
until delegate_pools are limited per provider it is not safe to join them, since you can get pretty large production and it won't work, you can
- limit the number of delegate pools per provider
- paginate this production and return only 20 rows of production (most likely won't make any sense)
- get delegate pools with different api endpoint which is paginated